schema/encoding/jsonschema/benchmark_complex_test.go

```diff
@@ -0,0 +1,221 @@
+package jsonschema_test
+
+import (
+	"bytes"
+	"github.com/rs/rest-layer/schema"
+	"github.com/rs/rest-layer/schema/encoding/jsonschema"
+	"math"
+	"testing"
+	"time"
+)
+
+// Optional schema field
+func Optional(s schema.Field) schema.Field {
```

Comment:
These are general helpers. I actually have a library of validation help functions that have different defaults, I copied these over for expediency. I'm not concerned since they are in the _test framework and not for general use here. I can certainly remove it, it just made it easier to port the code over.